src/common/pick_address.cc

```diff
@@ -354,10 +355,11 @@ int pick_addresses(
   addrs->v.clear();
 
   unsigned addrt = (flags & (CEPH_PICK_ADDRESS_PUBLIC |
+			     CEPH_PICK_ADDRESS_PUBLIC_BIND |
 			     CEPH_PICK_ADDRESS_CLUSTER));
-  if (addrt == 0 ||
-      addrt == (CEPH_PICK_ADDRESS_PUBLIC |
-		CEPH_PICK_ADDRESS_CLUSTER)) {
+  if (std::popcount(addrt) != 1) {
```

Comment:
Might want to consider std::has_single_bit()?

Reply:
`std::has_single_bit` is C++20 which is not enabled for quincy.

Reply:
Argh, `std::popcount()` is also C++20 and that's why this backport doesn't compile!  I thought it was C++17 for some reason...

Reply:
but so is popcount()...

Reply:
Yeah -- our comments raced ;)  Whichever Radek chooses to use in the main PR, the backport is going to need to resort to a builtin.